input/docs/handbook/collections/index.md

```diff
@@ -82,14 +82,124 @@ var myBindingOperation = mySource
 
 The API for the above is the same for cache and list.
 
-## So what's the difference between an observable list and an observable cache
+## So what's the difference between a SourceList and a SourceCache
 
-I get asked this question a lot and the answer is really simple.  If you have a unique id, you should use
-an observable cache as it is dictionary based which will ensure no duplicates can be added and it notifies on adds, updates and removes, whereas list allows duplicates and only has no concept of an update.
+If you have a unique id, you should use an observable cache as it is dictionary based which will ensure no duplicates can be added and it notifies on adds, updates and removes, whereas list allows duplicates and only has no concept of an update.
 
 There is another difference. The cache side of dynamic data is much more mature and has a wider range of operators. Having more operators is mainly because I found it easier to achieve good all round performance with the key based operators and do not want to add anything to Dynamic Data which inherently has poor performance.
 
-## How to track changes in collections of Reactive Objects?
+# Using DynamicData with ReactiveUI
+
+When building applications with ReactiveUI and DynamicData, you have a choice to work with mutable or with immutable collections. When working with immutable ones, using an `ObservableAsPropertyHelper<T>` is enough in simple cases. The `ObservableAsPropertyHelper<T>` represents an `Observable<T>`, a stream of values over time. You can treat those values as events, and the new values as event arguments. This means if you are using immutable collections, you can treat them as event arguments and update a property with a new collection each time it changes.
+
+### Simple Scenario
+
+```cs
+public class SimpleService 
+{
```

Comment:
The SimpleService has a number of things that wouldn't recommend it in most applications. It's probably best to remove this example. Eg I can't think of many scenarios where this is better apart from maybe where you know your enumerable lists aren't going to change often and they change dramatically.

Reply:
So the issues I can see
1. Performance, replacing items every time in 90% of scenarios could be an issue. I could maybe see it useful in REST based scenarios but I would do a Observable.Create() in those scenarios I've done something like 
```cs
    var dataSourceChanged = this.WhenAnyValue(x => x.DataSource);
     dataSourceChanged.Select(_ =>
                ObservableChangeSet.Create<TItem>(list => getItemsCommand.Subscribe(x => list.EditDiff(x)))
                    .Sort(SortExpressionComparer<TItem>.Descending(p => p.Id), SortOptions.UseBinarySearch)
                    .ObserveOnDispatcher()
                    .Bind(out this.items))
                .Switch()
                .Subscribe()
                .DisposeWith(this.disposableCombine);
```

This will create a re-generate the changeset if the model changes.

Reply:
I've also done this type of thing in the past with REST services

```cs
  private SourceCache<TItem, string> items = new SourceCache<TItem, string>(x => x.Id);

            getItemsCommand.Subscribe(list => this.items.EditDiff(list, (x, y) => x.Id.Equals(y.Id, StringComparison.OrdinalIgnoreCase)));
```

So the above, the EditDiff() will compare the values that come back from the `ReactiveCommand` and only adds item that are different.